lib/storage/src/content_manager/shard_distribution.rs

```diff
@@ -67,25 +66,48 @@ impl ShardDistributionProposal {
     pub fn local_shards_for(&self, peer_id: PeerId) -> Vec<ShardId> {
         self.distribution
             .iter()
-            .filter_map(
-                |(shard, peer)| {
-                    if peer == &peer_id {
-                        Some(*shard)
-                    } else {
-                        None
-                    }
-                },
-            )
+            .filter_map(|(shard, peers)| {
+                if peers.contains(&peer_id) {
+                    Some(shard)
+                } else {
+                    None
+                }
+            })
+            .copied()
             .collect()
     }
 
-    pub fn remote_shards_for(&self, peer_id: PeerId) -> Vec<(ShardId, PeerId)> {
+    pub fn remote_shards_for(&self, peer_id: PeerId) -> Vec<(ShardId, Vec<PeerId>)> {
         self.distribution
             .iter()
-            .filter(|(_shard, peer)| peer != &peer_id)
-            .copied()
+            .filter(|(_shard, peers)| !peers.contains(&peer_id))
+            .cloned()
             .collect()
     }
+
+    pub fn into(self, this_peer_id: PeerId) -> CollectionShardDistribution {
+        CollectionShardDistribution {
+            shards: self
+                .distribution
+                .into_iter()
+                .map(|(shard_id, peers)| match &peers[..] {
+                    [peer] if *peer == this_peer_id => (shard_id, ShardType::Local),
+                    [peer] => (shard_id, ShardType::Remote(*peer)),
```

Comment:
Is it possible to have a `ReplicaSet` with a single replicas or do we enforce a minimum replication factor of 2?

Reply:
For now based on our code-paths - no we can't have a replica set with 1 peer. And yes, that's what my initial thoughts were about the usage of the ReplicaSet struct. I'll add a comment